torch_xla/experimental/stablehlo_saved_model.py

```diff
@@ -0,0 +1,110 @@
+import copy
+import shutil
+import os
+import re
+import numpy as np
+import torch
+from torch import nn
+import torch_xla
+from torch_xla.core import xla_model as xm
+import tensorflow as tf
+import torchvision
+import torch._dynamo as torchdynamo
+
+from tensorflow.compiler.tf2xla.python import xla as tfxla  # type: ignore[import]
+
+from typing import Tuple, Type, Callable
+
+import sys
+
+
+class SHLOModel:
+
+  def __init__(self, model_bytecode, pos_to_orig_pos, pos_to_param,
+               sample_output):
+    self.model_bytecode = model_bytecode
+    self.pos_to_orig_pos = pos_to_orig_pos
+    self.pos_to_param = pos_to_param
+    self._total_number = len(pos_to_orig_pos) + len(pos_to_param)
+    output_tf = tf.convert_to_tensor(sample_output)
```

Comment:
curious why sample_output is needed. if they only need to know the shape, we can just create a fake tensor using `output.size` and `output.dtype`. Running the model on CPU once just to get the output is a bit wasteful

Reply:
Sure. Although the user probably won't know output shape / dtype without first running it.